core/psh/runfile/runfile.c

```diff
@@ -65,13 +64,13 @@ int psh_runfile(int argc, char **argv)
 
 		_psh_exit(EXIT_FAILURE);
 	}
-
-	waitpid(pid, NULL, 0);
+	if (pid > 0)
```

Comment:
```suggestion
	else {
```

Reply:
For MISRA sake (vfork() should be also removed from the first if)

Reply:
applied, thanks!